src/Streamly/Internal/Data/Stream/Async/Channel.hs

```diff
@@ -35,19 +34,25 @@ module Streamly.Internal.Data.Stream.Async.Channel
     -- * Stream operations
     , evalWithK
     , appendWithK
-    , concatMapWithK
+    , interleaveWithK
+    , concatMapWithK -- appendMap
+    , concatMapInterleaveWithK -- interleaveMap
```

Comment:
Append-ish
- `withChannelK` (Rename this to withAppendWithK. Basically indicate append-ish)
- `appendWithK`
- `evalWithK` (Rename this to indicate append-ish)
- `concatMapWithK` -- appendMap

Interleave-ish
- `withInterleaveChannelK`
- `interleaveWithK`
- `concatMapInterleaveWithK` -- interleaveMap

Reply:
Or is the default append-ish.

Reply:
Yes the default is append, to keep the default name shorter. I considered naming explicitly for both, but was not sure.